views-jstachio/src/main/java/io/micronaut/views/jstachio/JstachioViewResolver.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    @Override
+    public Optional<String> resolveView(HttpRequest<?> request, HttpResponse<?> response) {
+        Object body = response.body();
+        return body == null ?
+            Optional.empty() :
+            Optional.ofNullable(viewForClass.computeIfAbsent(body.getClass(), c -> resolveView(c).orElse(null)));
+    }
+
+    private Optional<String> resolveView(Class<?> bodyType) {
+        try {
+            BeanIntrospection<?> introspection = BeanIntrospection.getIntrospection(bodyType);
+            if (introspection.hasAnnotation(JStache.class)) {
+                AnnotationValue<JStache> jStacheAnnotation = introspection.getAnnotation(JStache.class);
+                Optional<AnnotationValue<JStacheConfig>> jStacheConfigAnnotationValueOptional = resolveJStacheConfig(introspection);
+                if (jStacheConfigAnnotationValueOptional.isEmpty()) {
+                    return jStacheAnnotation.stringValue(MEMBER_PATH);
+                }
+                return jStacheConfigAnnotationValueOptional.flatMap(jStacheConfigAnnotationValue -> resolveView(jStacheAnnotation, jStacheConfigAnnotationValue));
+            }
+        } catch (IntrospectionException e) {
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("{} may not be annotated with @Introspected", bodyType.getSimpleName(), e);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @NonNull
+    private Optional<AnnotationValue<JStacheConfig>> resolveJStacheConfig(@NonNull BeanIntrospection<?> introspection) {
+        if (introspection.hasAnnotation(JStacheConfig.class)) {
+            AnnotationValue<JStacheConfig> jStacheConfig = introspection.getAnnotation(JStacheConfig.class);
+            Optional<Class<?>> usingClassOptional = jStacheConfig.classValue(MEMBER_USING);
+            if (usingClassOptional.isPresent()) {
+                Class<?> usingClass = usingClassOptional.get();
+                try {
+                    BeanIntrospection<?> usingIntrospection = BeanIntrospection.getIntrospection(usingClass);
+                    if (usingIntrospection.hasAnnotation(JStacheConfig.class)) {
+                        return usingIntrospection.findAnnotation(JStacheConfig.class);
+                    }
+                } catch (IntrospectionException e) {
+                    if (LOG.isTraceEnabled()) {
+                        LOG.trace("Using class {} may not be annotated with @Introspected", usingClass.getSimpleName(), e);
+                    }
+                }
+            }
+        }
+        Class<?> beanType = introspection.getBeanType();
+        return resolveJStacheConfig(beanType);
+    }
+
+    @NonNull
+    private Optional<AnnotationValue<JStacheConfig>> resolveJStacheConfig(@NonNull Class<?> beanType) {
+        Package pkg = beanType.getPackage();
+        if (pkg != null) {
+            Optional<BeanConfiguration> beanConfigurationOptional = beanContext.findBeanConfiguration(pkg.getName());
```

Comment:
@graemerocher is this the best way to the the annotation metadata of `package-info.java`?